packages/cadl-ranch-specs/cadl-ranch-summary.md

```diff
@@ -1412,6 +1412,50 @@ Scenario that returns a different file encoding depending on the accept header.
 - image/png return a png image
 - image/jpeg return a jpeg image
 
+### Payload_Pageable_list
+
+- Endpoint: `get /payload/pageable`
+
+List users.
+
+SDK may hide the "maxpagesize" from API signature. The functionality of "maxpagesize" could be in related language Page model.
+
```

Comment:
@mikekistler Could you help review this test case? And I have below two questions.

- Server side may not support the `maxpagesize` so even we passed the value for this param the returned page size may or may not follow that, right?
- Regarding the client side if the tsp defined this query param, should we take this param as a regular one or hide it per language's implementation?

Reply:
If the service does not support `maxpagesize` then it should not be defined in the REST API. And if a client sends a query parameter that is not recognized/supported by the service, the service should fail the request with a 400 response code.

If the service does support `maxpagesize` and defines in the REST API, I don't know why we would hide it in the client library.

Reply:
@weidongxu-microsoft  Not sure the summary `SDK may hide the "maxpagesize" from API signature.` is proper?

Reply:
Mary, here is what Java does

API signature
```java
PagedIterable<User> list()
```

Hence `hide the "maxpagesize" from API signature` in summary. I use "may" because it may be language dependent.

Here the impl
```java
        return new PagedIterable<>(
                (pageSize) -> {
                    RequestOptions requestOptionsLocal = requestOptions == null ? new RequestOptions() : requestOptions;
                    if (pageSize != null) {
                        requestOptionsLocal.addRequestCallback(
                                requestLocal -> {
                                    UrlBuilder urlBuilder = UrlBuilder.parse(requestLocal.getUrl());
                                    urlBuilder.setQueryParameter("maxpagesize", String.valueOf(pageSize));
                                    requestLocal.setUrl(urlBuilder.toString());
                                });
                    }
                    return listSinglePage(requestOptionsLocal);
                },
                (nextLink, pageSize) -> {
                    RequestOptions requestOptionsLocal = new RequestOptions();
                    requestOptionsLocal.setContext(requestOptionsForNextPage.getContext());
                    if (pageSize != null) {
                        requestOptionsLocal.addRequestCallback(
                                requestLocal -> {
                                    UrlBuilder urlBuilder = UrlBuilder.parse(requestLocal.getUrl());
                                    urlBuilder.setQueryParameter("maxpagesize", String.valueOf(pageSize));
                                    requestLocal.setUrl(urlBuilder.toString());
                                });
                    }
                    return listNextSinglePage(nextLink, requestOptionsLocal);
                });
```

It is pretty long, but the idea is that it sets the "maxpagesize" from `pageSize` of `PagedIterable` (when available).

If TypeSpec of Swagger of the API does not have "maxpagesize" query parameter, none of the `maxpagesize` code will be generated, hence SDK won't pass "maxpagesize" to server.

Reply:
Got it! Thanks!